featuretools/tests/synthesis/test_dfs_method.py

```diff
@@ -53,6 +56,28 @@ def datetime_es():
     return datetime_es
 
 
+def test_dfs_empty_features():
+    error_text = 'No features can be generated from the specified primitives. Please make sure the primitives you are using are compatible with the variable types in your data.'
+    teams = pd.DataFrame({
+        'id': range(3),
+        'name': ['Breakers', 'Spirit', 'Thorns']
+    })
+    games = pd.DataFrame({
+        'id': range(5),
+        'home_team_id': [2, 2, 1, 0, 1],
+        'away_team_id': [1, 0, 2, 1, 0],
+        'home_team_score': [3, 0, 1, 0, 4],
+        'away_team_score': [2, 1, 2, 0, 0]
+    })
+    dataframes = {'teams': (teams, 'id', None, {'name': 'natural_language'}), 'games': (games, 'id')}
+    relationships = [('teams', 'id', 'games', 'home_team_id')]
+    features = dfs(dataframes, relationships, target_dataframe_name="teams", features_only=True)
+    assert (isinstance(features, list) and features != []), True
```

Comment:
I'd like the patch of `DeepFeatureSynthesis` returning `[]` to also apply to this run dfs / assert code so we can test that an empty feature list is fine if `features_only=True`

we'd need to separate the `with` block into two blocks: the patch context which would apply to both `dfs` calls, and the `pytest.raises` with block only applying to the `features_only=False` call

Reply:
patch DeepFeatureSynthesis to return []